packages/venia-ui/lib/components/Header/cartTrigger.js

```diff
@@ -1,75 +1,67 @@
-import React, { Component } from 'react';
-import { connect } from '@magento/venia-drivers';
-import { compose } from 'redux';
-import PropTypes from 'prop-types';
+import React, { useEffect, useMemo } from 'react';
+import { func, number, shape, string } from 'prop-types';
+import { ShoppingCart as ShoppingCartIcon } from 'react-feather';
 
-import { toggleCart } from '../../actions/cart';
+import Icon from '../Icon';
 import CartCounter from './cartCounter';
 
-import Icon from '../Icon';
-import { ShoppingCart as ShoppingCartIcon } from 'react-feather';
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import defaultClasses from './cartTrigger.css';
 
-export class Trigger extends Component {
-    static propTypes = {
-        children: PropTypes.node,
-        classes: PropTypes.shape({
-            root: PropTypes.string
-        }),
-        toggleCart: PropTypes.func.isRequired,
-        itemsQty: PropTypes.number
+const getCartIcon = cartDetails => {
+    const itemsQty = cartDetails.items_qty;
+    const iconColor = 'rgb(var(--venia-text))';
+    const svgAttributes = {
+        stroke: iconColor
     };
 
-    get cartIcon() {
-        const {
-            cart: { details }
-        } = this.props;
-        const itemsQty = details.items_qty;
-        const iconColor = 'rgb(var(--venia-text))';
-        const svgAttributes = {
-            stroke: iconColor
-        };
+    if (itemsQty > 0) {
+        svgAttributes.fill = iconColor;
+    }
 
-        if (itemsQty > 0) {
-            svgAttributes.fill = iconColor;
-        }
+    return <Icon src={ShoppingCartIcon} attrs={svgAttributes} />;
+};
 
-        return <Icon src={ShoppingCartIcon} attrs={svgAttributes} />;
-    }
+const Trigger = props => {
+    const { cart, getCartDetails, toggleCart } = props;
+    const { details: cartDetails } = cart;
+    const { items_qty: numItems } = cartDetails;
 
-    render() {
-        const {
-            classes,
-            toggleCart,
-            cart: { details }
-        } = this.props;
-        const { cartIcon } = this;
-        const itemsQty = details.items_qty;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        return (
-            <button
-                className={classes.root}
-                aria-label="Toggle mini cart"
-                onClick={toggleCart}
-            >
-                {cartIcon}
-                <CartCounter counter={itemsQty ? itemsQty : 0} />
-            </button>
-        );
-    }
-}
+    useEffect(() => {
+        if (getCartDetails) {
```

Comment:
Is this a necessary conditional? Seems like the function should always be present.

Reply:
Yeah I wasn't really sure about this. It will always be present because I connected it that way, but I was thinking if someone else uses the `Header` component they may or may not connect it.

Marking this prop type as `.isRequired` is along these same lines.

I didn't mark it as `isRequired`, so I included this conditional.

Reply:
`Header` is [available as an export](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/Header/index.js#L1) from `Header/index.js`.

Reply:
But `Header` passes the function now, so it _should_ always be defined.

I'd probably just mark it as `isRequired` and be done with.

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/1556/commits/8bb2e7e3ff3feadfb28dbde4400f2a98d372536d.